### DISCUSSION TITLE
Am I using `narrow_symbols` incorrectly? Cannot get uniform icons

### DISCUSSION THREAD

**Discussion by benfrain** [kovidgoyal/kitty#7003] [Q&A] [answered]
Firstly, I apologise @kovidgoyal as I know this is a conversation you have had a few times ;)

I have the following two sets of narrow_symbols in my Kitty config:

```
narrow_symbols U+23FB-U+23FE,U+2665,U+26A1,U+2B58,U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0C8,U+E0CA,U+E0CC-U+E0D2,U+E0D4,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E634,U+E700-U+E7C5,U+EA60-U+EBEB,U+F000-U+F2E0,U+F300-U+F32F,U+F400-U+F4A9,U+FC1F,U+F918,U+F500-U+F8FF 2 

# mapped from the symbols here https://github.com/onsails/lspkind.nvim/blob/master/lua/lspkind/init.lua and codes retrieved from https://codepoints.net/
narrow_symbols U+F027F,U+F01A7,U+F0295,U+F423,U+F0722,U+F002B,U+F0831,U+F0E8,U+F487,U+F0722,U+F046D,U+F03A0,U+F15D,U+F030B,U+F44F,U+F03D8,U+F0219,U+F0207,U+F024B,U+F15D,U+F03FF,U+F0645,U+F0E7,U+F0195,U+EA93,U+EA8C,U+EA8C,U+EA8C,U+EB5F,U+EA88,U+EB5B,U+EB61,U+EA8B,U+EB65,U+EA96,U+EA95,U+EB62,U+EB66,U+EB5C,U+EA7B,U+EA94,U+EA83,U+EA95,U+EB5D,U+EA91,U+EA86,U+EB64,U+EA92 2
```

However, for the same icons in the same menu of Neovim, I am then seeing a disparity in the width of the same glyph:

<img width="567" alt="image" src="https://github.com/kovidgoyal/kitty/assets/1318466/965abdaa-9df2-45ae-9eb5-10b37f3b8535">

I thought the `2` on the end would make those symbols span two cells? 

If not, what would the reason be that those symbols are not the same width in the same context? Just some oddity in Neovim?

I have tried setting those two `narrow_symbol` instances to have a cell width of 1, which solves the issue in my completion menu (albeit makes them a little small) - Woohoo!:

<img width="695" alt="image" src="https://github.com/kovidgoyal/kitty/assets/1318466/02dc601e-1417-4f4a-a77a-454d61811802">

However, I then have incredibly small icons in my gutter - Sad Trombone:

<img width="332" alt="image" src="https://github.com/kovidgoyal/kitty/assets/1318466/58bb32d1-d701-41b1-9a35-f75a49056a6e">

Instead of the size I would expect to see them:

<img width="395" alt="image" src="https://github.com/kovidgoyal/kitty/assets/1318466/eec3a0ed-2caf-4e6c-8690-1cdbd5ebe704">

I also ran `--debug-font-fallback` and this was the output:

```
 kitty --debug-font-fallback
[016 21:31:23.137286] Preloaded font faces:
[016 21:31:23.137307] normal face: /Users/benfrain/Library/Fonts/VictorMono-Regular.otf
[016 21:31:23.137313] bold face: /Users/benfrain/Library/Fonts/VictorMono-SemiBold.otf
[016 21:31:23.137317] italic face: /Users/benfrain/Library/Fonts/VictorMono-Italic.otf
[016 21:31:23.137321] bi face: /Users/benfrain/Library/Fonts/VictorMono-SemiBoldItalic.otf
[016 21:31:23.137326] Symbol map faces:
[016 21:31:23.137329] /Users/benfrain/Library/Fonts/SymbolsNerdFontMono-Regular.ttf
U+276f Face(family=Menlo, full_name=Menlo Regular, postscript_name=Menlo-Regular, path=/System/Library/Fonts/Menlo.ttc, units_per_em=2048, ascent=35.3, descent=9.0, leading=0.0, point_sz=0.0, scaled_point_sz=38.0, underline_position=-2.4 underline_thickness=1.7) (new face)
U+276f Face(family=Menlo, full_name=Menlo Regular, postscript_name=Menlo-Regular, path=/System/Library/Fonts/Menlo.ttc, units_per_em=2048, ascent=35.3, descent=9.0, leading=0.0, point_sz=0.0, scaled_point_sz=38.0, underline_position=-2.4 underline_thickness=1.7)
```

**Comment by kovidgoyal** [comment, marked as answer]
narrow_symbols only narrow symbols that would otherwise be rendered
wide. If a symbol would not otherwise be rendered wide it has no effect. 
You likely need to follow up the symbols that are narrow with the VS16
modifier to make them wide as their default appearance in the unicode
standard will be text not emoji.

**Reply by benfrain** [reply]
Just for any future travellers - I ended up finding that was one instance (diagnostics) using some nerd font glyphs that didn't go to two cells wide when space was available. I used the nerd font cheat sheet: https://www.nerdfonts.com/cheat-sheet to search for a comparable icon for each instance that did go bigger with a space after and now all is good. Thanks @kovidgoyal
